modules/rds/main.tf

```diff
@@ -1,84 +1,172 @@
+  tags                                = local.tags
 }
 
+
+## =============================================================================
+## NULL RESOURCES - 1x Use
+## =============================================================================
+
+## bootstrap - Bootstrap lambda that creates/modifies database objects on deploys
+## =============================================================================
+# https://registry.terraform.io/providers/hashicorp/null/latest/docs/resources/resource
+# https://www.terraform.io/docs/language/resources/provisioners/local-exec.html
 resource "null_resource" "bootstrap" {
+  # Determine what has to change to trigger this resource being created/re-created.
   triggers = {
-    bootstrap_lambda_hash = var.lambda_source_code_hash
+    bootstrap_lambda_hash = var.db_deploy_source_code_hash
   }
 
+  # Execute the db_deploy lambda 1 time if the resource is created/re-created.
   provisioner "local-exec" {
-    command = "aws lambda invoke --function-name ${var.db_lambda_deploy_arn} ${local.used_profile} --region ${var.region} 'invoke-response.out'"
+    command = "aws lambda invoke --function-name ${var.db_deploy_arn} ${local.used_profile} --region ${var.region} 'db_deploy-response.out'"
   }
 
   depends_on = [aws_db_instance.postgresql]
 }
 
+
+## TODO: Should create null resource to handle password changes ORCA-145
+
+
+## =============================================================================
+## SECRET MANAGER RESOURCES
+## =============================================================================
+
+## drdb-admin-pass - postgres (root) user password
+## =============================================================================
 resource "aws_secretsmanager_secret" "drdb-admin-pass" {
-  name = "${var.prefix}-drdb-admin-pass"
-  description = "Admin password to be used for the PostgreSQL DB"
-  tags        = var.default_tags
+  ## OPTIONAL
+  description             = "Admin password to be used for the PostgreSQL DB"
+  name                    = "${var.prefix}-drdb-admin-pass"
+  recovery_window_in_days = 0
+  tags                    = local.tags
 }
 
+
 resource "aws_secretsmanager_secret_version" "drdb-admin-pass" {
-  secret_id     = aws_secretsmanager_secret.drdb-admin-pass.id
+  ## REQUIRED
+  secret_id = aws_secretsmanager_secret.drdb-admin-pass.id
+
+  ## OPTIONAL
   secret_string = var.postgres_user_pw
 }
 
+
+## drdb-user-pass - Application user database password
+## =============================================================================
 resource "aws_secretsmanager_secret" "drdb-user-pass" {
-  name = "${var.prefix}-drdb-user-pass"
-  description = "User password to be used for the PostgreSQL DB"
-  tags        = var.default_tags
+  ## OPTIONAL
+  description             = "Application user password to be used for the PostgreSQL DB"
+  name                    = "${var.prefix}-drdb-user-pass"
+  recovery_window_in_days = 0
```

Comment:
also same for recovery_window_in_days. How about we move it to variables.tf?

Reply:
I would like to keep recovery_window_in_days set to 0. This value was unset in the original and the 30 day window causes issues especially if you want to tear down your stack and re-deploy. In the original, if you tore down your stack, you had to manually go remove secrets through the dashboard which allowed you to go from a 30 day to a 7 day wait time. I don't feel this is something we should bubble up to the users but should just set for them.